cmd/otel-allocator/server/server.go

```diff
@@ -93,38 +85,36 @@ func (s *Server) Shutdown(ctx context.Context) error {
 	return s.server.Shutdown(ctx)
 }
 
-// ScrapeConfigsHandler returns the available scrape configuration discovered by the target allocator.
+// UpdateScrapeConfigResponse updates the scrape config response and potential error resulting from an update.
 // The target allocator first marshals these configurations such that the underlying prometheus marshaling is used.
 // After that, the YAML is converted in to a JSON format for consumers to use.
-func (s *Server) ScrapeConfigsHandler(c *gin.Context) {
-	configs := s.discoveryManager.GetScrapeConfigs()
-
-	hash, err := hashstructure.Hash(configs, nil)
+func (s *Server) UpdateScrapeConfigResponse(configs map[string]*promconfig.ScrapeConfig) {
+	var configBytes []byte
+	configBytes, err := yaml.Marshal(configs)
 	if err != nil {
-		s.logger.Error(err, "failed to hash the config")
-		s.errorHandler(c.Writer, err)
+		s.scrapeConfigErr = err
```

Comment:
IMO we should return an error from here rather than store it, the discovery manager's method already has error handling baked in. This also means that if we see some type of bogus configuration here, we can bubble that up to the caller while still holding on to the last good configuration. In this case we'd want to store the hash only after a successful call to `UpdateScrapeConfigResponse`.

Reply:
Sounds right to me, I initially wanted to keep behavior similar to what we have now, but returning last known good config is preferable to me as well.